code/modules/reagents/reagent_containers/cups/glassbottle.dm

```diff
@@ -579,13 +579,17 @@
 			isGlass = FALSE
 	return
 
+/obj/item/reagent_containers/cup/glass/bottle/molotov/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..(hit_atom, throwingdatum, do_splash = FALSE)
+
 /obj/item/reagent_containers/cup/glass/bottle/molotov/smash(atom/target, mob/thrower, ranged = FALSE)
 	var/firestarter = 0
 	for(var/datum/reagent/contained_reagent in reagents.reagent_list)
 		for(var/accelerant_type in accelerants)
 			if(istype(contained_reagent, accelerant_type))
 				firestarter = 1
 				break
+	SplashReagents(target, TRUE)
```

Comment:
This is already handled in parent

Reply:
This being handled in the parent is bad, because then the IF does not have any reagents to execute, so delaying the Splash Reagents is important, unless there's some better way like saving the reagent state temporarily.

Reply:
And, as you can see it is not being handled in the parent, because the parent isn't being called, it is being replaced. No ..() call

Reply:
Removed.